rosetta-maven-plugin/src/main/java/com/regnosys/rosetta/maven/ResourceFormatterMojo.java

```diff
@@ -0,0 +1,201 @@
+
+		FormattingOptions formattingOptions = null;
+		try {
+			formattingOptions = readFormattingOptions(formattingOptionsPath);
+		} catch (IOException e) {
+			LOGGER.error("Config file not found.", e);
+		}
+
+		Injector inj = new RosettaStandaloneSetup().createInjectorAndDoEMFRegistration();
+		ResourceSet resourceSet = inj.getInstance(ResourceSet.class);
+		ResourceFormatterService formatterService = inj.getInstance(ResourceFormatterService.class);
+
+		List<Resource> resources;
+		try {
+			// Find all .rosetta files in the directory and load them from disk
+			resources = Files.walk(directory).filter(path -> path.toString().endsWith(".rosetta"))
+					.map(file -> resourceSet.getResource(URI.createFileURI(file.toString()), true))
+					.collect(Collectors.toList());
+		} catch (IOException e) {
+			throw new MojoFailureException("Error processing files: " + e.getMessage(), e);
+		}
+		// format resources
+		formatterService.formatCollection(resources, createPreferences(formattingOptions),
+				(resource, formattedText) -> {
+					Path resourcePath = Path.of(resource.getURI().toFileString());
+					try {
+						Files.writeString(resourcePath, formattedText);
+						LOGGER.info("Content written to file: " + resourcePath);
+					} catch (IOException e) {
+						LOGGER.error("Error writing to file.", e);
+					}
+				});
+	}
+
+	public static FormattingOptions readFormattingOptions(String optionsPath) throws IOException, MojoFailureException {
+		InputStream resourceStream;
+		// If path not given, use default one
+		if (optionsPath == null) {
+			// Retrieve resource as an InputStream
+			resourceStream = ResourceLoader.class.getClassLoader().getResourceAsStream(DEFAULT_FORMATTING_OPTIONS_PATH);
+		} else {
+			resourceStream = new FileInputStream(optionsPath);
+		}
+
+		// Create an ObjectMapper, read JSON into a Map
+		ObjectMapper objectMapper = new ObjectMapper();
+		Map<String, Object> map = null;
+		try {
+			map = objectMapper.readValue(resourceStream, Map.class);
+		} catch (IOException e) {
+			throw new MojoFailureException("Error processing formatting options: " + e.getMessage(), e);
+		}
+
+		// Create a FormattingOptions object
+		FormattingOptions formattingOptions = new FormattingOptions();
+
+		// Populate the FormattingOptions object
+		for (Map.Entry<String, Object> entry : map.entrySet()) {
+			String key = entry.getKey();
+			Object value = entry.getValue();
+
+			if (value instanceof String) {
+				formattingOptions.putString(key, (String) value);
+			} else if (value instanceof Number) {
+				formattingOptions.putNumber(key, (Number) value);
+			} else if (value instanceof Boolean) {
+				formattingOptions.putBoolean(key, (Boolean) value);
+			} else {
+				throw new IllegalArgumentException("Unsupported value type for key: " + key);
+			}
+		}
+		return formattingOptions;
+	}
+
+	public static ITypedPreferenceValues createPreferences(FormattingOptions options) {
```

Comment:
Since this code is exactly the same as in the IDE project, I think it would be good to refactor it so we use some common code. How about adding a `FormattingOptionsAdaptor` in `rosetta-lang` which does the conversion, and then that one can be injected both in IDE and in this plugin?

Reply:
You can then probably move the lsp4j dependency from this project to rosetta-lang as well.